src/lib/y2partitioner/immediate_unmount.rb

```diff
@@ -0,0 +1,115 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast/i18n"
+require "yast2/popup"
+require "y2partitioner/device_graphs"
+
+module Y2Partitioner
+  # Mixin
+  module ImmediateUnmount
+  private
+
+    extend Yast::I18n
+
+    # Sets textdomain
+    def included(_target)
+      textdomain "storage"
+    end
+
+    def immediate_unmount(device, full_message: nil, note: nil, allow_continue: true)
+      loop do
+        case confirm_unmount(device, full_message, note, allow_continue)
+        when :unmount
+          unmount(device) || next
+        when :continue
+          break true
+        when :cancel
+          break false
+        end
+      end
+    end
+
+    def confirm_unmount(device, full_message, note, allow_continue)
+      headline = Yast::Label.WarningMsg
+
+      message = immediate_unmount_message(device, full_message, note, allow_continue)
+
+      buttons = inmmediate_unmount_buttons(allow_continue)
+
+      Yast2::Popup.show(message, headline: headline, buttons: buttons, focus: :cancel)
+    end
+
+    def unmount(device)
+      device.mount_point.immediate_deactivate
+      true
+    rescue Storage::Exception
```

Comment:
I would really like to see at least logging of exception if not passing it to show_unmount_error so user can see some details why unmount failed.